File: frontend/src/types/characterTagType.ts
/**
 * Reprezentuje typ tagu pro charakter (např. NPC, Player Controlled, Merchant).
 */
export interface CharacterTagType {
  id: number;
  world_id: number;
  name: string;
  created_at: string; // ISO date string
  updated_at: string; // ISO date string
}

// Typ pro vytváření nového typu tagu
export interface CharacterTagTypeCreate {
  name: string;
}

// Typ pro aktualizaci typu tagu
export interface CharacterTagTypeUpdate {
  name?: string;
} 

File: backend/src/app/schemas/character_tag_type.py
from pydantic import BaseModel
from datetime import datetime

# Základní schéma pro data, která se vrací
class CharacterTagTypeBase(BaseModel):
    name: str

# Schéma pro vytváření
class CharacterTagTypeCreate(CharacterTagTypeBase):
    pass

# Schéma pro aktualizaci
class CharacterTagTypeUpdate(CharacterTagTypeBase):
    name: str | None = None # Všechny fieldy jsou volitelné při aktualizaci

# Schéma pro data uložená v DB (pro čtení)
class CharacterTagTypeInDBBase(CharacterTagTypeBase):
    id: int
    world_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True # Umožní mapování z SQLAlchemy modelu

# Finální schéma pro čtení dat z API
class CharacterTagType(CharacterTagTypeInDBBase):
    pass 